core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala

```diff
@@ -207,9 +207,16 @@ private[deploy] class SparkSubmitArguments(args: Seq[String], env: Map[String, S
       uriScheme match {
         case "file" =>
           try {
-            val jar = new JarFile(uri.getPath)
-            // Note that this might still return null if no main-class is set; we catch that later
-            mainClass = jar.getManifest.getMainAttributes.getValue("Main-Class")
+            var jar: JarFile = null
+            Utils.tryWithSafeFinally {
```

Comment:
Ah pardon, I meant `Utils.tryWithResource`. This one doesn't help close the resource. I don't know which one is actually clearer; up to you.